Update openai readme, clarify which metrics are available based on setup method (#16229)
DataDog/integrations-core: openai/README.md

<!-- xxx tabs xxx -->
<!-- xxx tab "Python" xxx -->

> **Note**: This setup method will not collect `openai.api.usage.*` metrics, unless you also follow the API key setup instructions.

Reviewer: ```suggestion
**Note**: This setup method does not collect `openai.api.usage.*` metrics. To collect these metrics, also follow the API key setup instructions.
```

<!-- xxz tab xxx -->
<!-- xxx tab "Node.js" xxx -->

> **Note**: This setup method will not collect `openai.api.usage.*` metrics, unless you also follow the API key setup instructions.

Reviewer: ```suggestion
**Note**: This setup method does not collect `openai.api.usage.*` metrics. To collect these metrics, also follow the API key setup instructions.
```

<!-- xxz tabs xxx -->
<!-- xxx tab "API Key" xxx -->

> **Note**: This setup method will only collect  `openai.api.usage.*` metrics. To collect all metrics provided by this integration, also follow the APM setup instructions.

Reviewer: ```suggestion
**Note**: This setup method only collects `openai.api.usage.*` metrics. To collect all metrics provided by this integration, also follow the APM setup instructions.
```

### Metrics

> **Note**: The `openai.api.usage.*` metrics are only collected with the API key setup method. All remaining metrics below are collected with the APM setup methods.

Reviewer: ```suggestion
The `openai.api.usage.*` metrics are only collected with the API key setup method. All remaining metrics below are collected with the APM setup methods.
```
